src/node.js

```diff
@@ -383,27 +403,10 @@
           tock = nextTickQueue[tickInfo[kIndex]++];
           callback = tock.callback;
           args = tock.args;
-          // Using separate callback execution functions helps to limit the
-          // scope of DEOPTs caused by using try blocks and allows direct
+          // Using separate callback execution functions allows direct
           // callback invocation with small numbers of arguments to avoid the
           // performance hit associated with using `fn.apply()`
-          if (args === undefined) {
-            nextTickCallbackWith0Args(callback);
-          } else {
-            switch (args.length) {
-              case 1:
-                nextTickCallbackWith1Arg(callback, args[0]);
-                break;
-              case 2:
-                nextTickCallbackWith2Args(callback, args[0], args[1]);
-                break;
-              case 3:
-                nextTickCallbackWith3Args(callback, args[0], args[1], args[2]);
-                break;
-              default:
-                nextTickCallbackWithManyArgs(callback, args);
-            }
-          }
+          _combinedTickCallback(args, callback);
```

Comment:
To remove a line in the call stack, think we could just move the if/else back into here? Open to why it should be kept in another function.